### PR TITLE
Clean-up sparse max/min argmax/argmin

### PR DESCRIPTION
**Pull request by emcastillo**
Removed duplicated code and intrincated ifs

### CODE REVIEW

File: cupyx/scipy/sparse/compressed.py [hunk 1]
```diff
@@ -893,82 +801,21 @@ def _minor_reduce(self, ufunc, axis, nonzero):
                 major_index.
 
         """
-
+        shape = self.shape[int(not(axis))]
```
Comment (toslunar) [inline cupyx/scipy/sparse/compressed.py:804]: - I found `1 - axis` in `cupyx/scipy/sparse/data.py`.
- For readability, could you give a more detailed name?  How about `shape` -> `out_shape` and `value` -> `out`?

Comment (emcastillo) [reply cupyx/scipy/sparse/compressed.py]: Good catch :).
Thanks!

**Review by toslunar — DISMISSED** [review]

**Comment by toslunar** [comment]
pfnCI, test this please.

**Comment by emcastillo** [comment]
Fixed jenkins

**Review by toslunar — DISMISSED** [review]

**Comment by emcastillo** [comment]
rebased because of the flake8 error :)

**Review by toslunar — APPROVED** [review]

**Comment by toslunar** [comment]
pfnCI, test this please.

**Comment by toslunar** [comment]
The Jenkins failure is irrelevant to the PR.

**Pull request merged**